app/javascript/packs/webauthn-setup.ts

```diff
@@ -4,19 +4,30 @@ import {
   extractCredentials,
   longToByteArray,
 } from '@18f/identity-webauthn';
+import { forceRedirect } from '@18f/identity-url';
+import type { Navigate } from '@18f/identity-url';
 
 /**
  * Reloads the current page, presenting the message corresponding to the given error key.
  *
  * @param error Error key for which to show message.
  * @param options Optional options.
  * @param options.force If true, reload the page even if that error is already shown.
+ * @param options.search Initial search value.
+ * @param options.navigate Navigate implementation.
  */
-export function reloadWithError(error: string, { force = false }: { force?: boolean } = {}) {
-  const params = new URLSearchParams(window.location.search);
+export function reloadWithError(
+  error: string,
+  {
+    force = false,
+    search = window.location.search,
+    navigate = forceRedirect,
+  }: { force?: boolean; search?: string; navigate?: Navigate } = {},
+) {
+  const params = new URLSearchParams(search);
   if (force || params.get('error') !== error) {
     params.set('error', error);
-    window.location.search = params.toString();
+    navigate(`?${params}`);
```

Comment:
would this option drop the `.hash` of the `window.location`? (or I guess, would it be worth worrying about if it did?)

Reply:
> would this option drop the `.hash` of the `window.location`? (or I guess, would it be worth worrying about if it did?)

Yeah, it would. And while I don't think it'd be an actual problem, it seems like the sort of thing where a bug could easily slip in.

In 02aa2fc, I reverted back to an implementation a little closer to what it was recently, while still allowing overriding.

Reply:
I would maybe have taken a separate approach instead of a separate callback? Like a `navigate` function that copies + replaces the `document.location`?

```js
const locationURL = new URL(window.location);
locationURL.search = params.toString();
navigate(locationURL.toString());

Reply:
Ah, took me a moment to follow what you're suggesting, but I think I see what you're saying now. Let me give that a shot.

Reply:
I think I got that working in d27283b.

Reply:
big fan! thx

Reply:
Took it a little further in a5d9da56c to avoid managing both `window.locatin.href` and `window.location.search`, and do a single `URL` constructor rather than both `URL` and `URLSearchParams.